CMakeLists.txt

```diff
@@ -532,6 +488,88 @@ while(library_variants_copy)
             ${INSTALL_DIR}/lib/${target} ${install_directory}/lib
     )
 
+
+    if(CMAKE_INSTALL_MESSAGE STREQUAL NEVER)
+        set(MESON_INSTALL_QUIET "--quiet")
+    endif()
+    if(libc_type STREQUAL "picolibc")
+        ExternalProject_Add(
+            picolibc_${variant}
+            SOURCE_DIR ${picolibc_SOURCE_DIR}
+            INSTALL_DIR ${install_directory}
+            PREFIX picolibc/${variant}
+            DEPENDS clang lld llvm-ar llvm-config llvm-nm llvm-ranlib llvm-strip compiler_rt_${variant}
+            CONFIGURE_COMMAND ${MESON_EXECUTABLE} -Dtests=true -Dincludedir=include -Dlibdir=lib --prefix <INSTALL_DIR> --cross-file <BINARY_DIR>/meson-cross-build.txt ${picolibc_SOURCE_DIR}
+            BUILD_COMMAND ninja
+            INSTALL_COMMAND meson install ${MESON_INSTALL_QUIET}
+            USES_TERMINAL_CONFIGURE TRUE
+            USES_TERMINAL_BUILD TRUE
+            USES_TERMINAL_TEST TRUE
+            LIST_SEPARATOR ,
+            # Always run the build command so that incremental builds are correct.
+            BUILD_ALWAYS TRUE
+            CONFIGURE_HANDLED_BY_BUILD TRUE
+        )
+        # Set meson_c_args to a comma-separated list of the clang path
+        # and flags e.g. 'path/to/clang', '--target=armv6m-none-eabi',
+        # '-march=armv6m'
+        set(meson_c_args "${common_flags}")
+        string(REPLACE " " "', '" meson_c_args "${meson_c_args}")
+        set(meson_c_args "'${llvm_bin}/clang', '${meson_c_args}'")
+        #replace -lgcc for picolibc
+        set(clang_libraries "'${LLVM_BINARY_DIR}/../compiler-rt/${variant}/install/lib/${target}/libclang_rt.builtins.a'")
+
+        ExternalProject_Get_Property(picolibc_${variant} BINARY_DIR)
+        configure_file(cmake/meson-cross-build.txt.in ${BINARY_DIR}/meson-cross-build.txt @ONLY)
+        set(libc_target picolibc_${variant})
+        set(libc_specific_runtimes_options
+            -DLIBCXXABI_ENABLE_EXCEPTIONS=OFF
+            -DLIBCXXABI_ENABLE_THREADS=OFF
+            -DLIBCXX_ENABLE_EXCEPTIONS=OFF
+            -DLIBCXX_ENABLE_MONOTONIC_CLOCK=OFF
+            -DLIBCXX_ENABLE_RANDOM_DEVICE=OFF
+            -DLIBCXX_ENABLE_RTTI=OFF
+            -DLIBCXX_ENABLE_THREADS=OFF
+            -DLIBCXX_ENABLE_WIDE_CHARACTERS=OFF
+            -DLIBUNWIND_ENABLE_THREADS=OFF
+        )
+        set(picolibc_tests printf_scanf
+                           printf-tests
+                           rand
+                           regex
+                           ungetc
+                           fenv
+                           long_double
+                           math_errhandling
+                           malloc
+                           tls
```

Comment:
I guess tls needs commenting out since you've disabled building it

Reply:
true